src/neurostatslib/basis.py

```diff
@@ -0,0 +1,621 @@
+"""Bases classes.
```

Comment:
why is this file double the length of the old `basis.py`? looks like it contains things that are in the `edoardo` branch -- the new spline bases. those should be removed so that we keep the two changes separate.

Reply:
I mention this in the PR request comment, the basis class will be rebuild from scratch, so ignore any diffs. (this version works, doesn't brake the code but still will be re-structured)

Reply:
reverted to the main and created a specific branch to tackle the basis function issue